source/browseMode.py

```diff
@@ -250,6 +250,13 @@ def _getLabelForProperties(self, labelPropertyGetter):
 			label = content
 		return label
 
+def getObjectID(obj):
+	try:
+		return obj.IA2UniqueID

```

Comment:
NVDAObjects should have a new 'uniqueID' property. For IA2 it should be self.windowHandle,self.ia2UniqueID, for UIA it should be self.UIAElement.getRuntimeId(). For generic IAccessible NVDAObjects it should be self.IAccessibleIdentity if it exists otherwise hash(self.IAccessibleObject). For Window it should be self.windowHandle.

Reply:
I just checked current head branch - NVDA objects don't have uniqueID property. Do you mean it will be added in the future or am I missing something? the In any case, this code is only needed to make sure that async part of code is called on the same NVDA object - which in all cases that I know of it is. So it is kind of a second line of defense - to identify async-related bugs early.

Reply:
I was requesting that you implement the uniqueID property on NVDAObjects as I described. If you fully trust this works without this check, then remove it. However, I cannot approve the code as is with specific mention to IAccessible2 in browseMode.py.

Reply:
Sorry for misunderstanding - I implemented .uniqueID property as you requested in IAccessible, UAI and Window classes. I couldn't find "generic IAccessible NVDAObjects " - could you point me where it is located? But now browseMode doesn't have any IA2-specific code any more.